### PR TITLE
[Rust] Checkout a version of dataset.

### PR DESCRIPTION
**Pull request by eddyxu**
* `Table::open()` from absolute path, and gives the responsibility of organizing metadata out of Table object
* Fix Clippy warnings
* Add `Table::checkout(version)` API

### CODE REVIEW

File: rust/vectordb/src/index/vector.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ pub trait VectorIndexBuilder {
     fn get_replace(&self) -> bool;
 }
 
+#[derive(Default)]
```
Comment (chebbyChefNEQ) [inline rust/vectordb/src/index/vector.rs:27]: I think the default of `bool` is `false` we might want to manually implement default here

File: rust/vectordb/src/table.rs [hunk 2]
```diff
@@ -69,25 +63,42 @@ impl Table {
...
     /// # Returns
     ///
     /// * A [Table] object.
-    pub async fn open_with_params(
-        base_uri: &str,
-        name: &str,
-        params: OpenTableParams,
-    ) -> Result<Self> {
-        let path = Path::new(base_uri);
+    pub async fn open_with_params(uri: &str, name: &str, params: &ReadParams) -> Result<Self> {
```
Comment (chebbyChefNEQ) [inline rust/vectordb/src/table.rs:71]: do we still need `name`?

Comment (eddyxu) [reply rust/vectordb/src/table.rs:71]: Yea, this is for the table to have a better display name without leaking storage details. 

Might be able to deduct the name from uri if not provided, wdyt @chebbyChefNEQ ?

File: rust/vectordb/src/index/vector.rs (hunk 1 above)
Comment (eddyxu) [reply rust/vectordb/src/index/vector.rs]: Fixed

**Review by chebbyChefNEQ — APPROVED** [review]

**Pull request merged**